sim/battle-stream.ts

```diff
@@ -169,58 +176,73 @@ export class BattleStream extends Streams.ObjectReadWriteStream<string> {
  * streams, for ease of consumption.
  */
 export function getPlayerStreams(stream: BattleStream) {
-	const omniscient = new Streams.ObjectReadWriteStream({
-		write(data: string) {
-			stream.write(data);
-		},
-		end() {
-			stream.end();
-		},
-	});
-	const spectator = new Streams.ObjectReadStream({
-		read() {},
-	});
-	const p1 = new Streams.ObjectReadWriteStream({
-		write(data: string) {
-			stream.write(data.replace(/(^|\n)/g, `$1>p1 `));
-		},
-	});
-	const p2 = new Streams.ObjectReadWriteStream({
-		write(data: string) {
-			stream.write(data.replace(/(^|\n)/g, `$1>p2 `));
-		},
-	});
+	const streams = {
+		omniscient: new Streams.ObjectReadWriteStream({
+			write(data: string) {
+				stream.write(data);
+			},
+			end() {
+				stream.end();
+			},
+		}),
+		spectator: new Streams.ObjectReadStream({
+			read() {},
+		}),
+		p1: new Streams.ObjectReadWriteStream({
+			write(data: string) {
+				stream.write(data.replace(/(^|\n)/g, `$1>p1 `));
+			},
+		}),
+		p2: new Streams.ObjectReadWriteStream({
+			write(data: string) {
+				stream.write(data.replace(/(^|\n)/g, `$1>p2 `));
+			},
+		}),
+		p3: new Streams.ObjectReadWriteStream({
+			write(data: string) {
+				stream.write(data.replace(/(^|\n)/g, `$1>p3 `));
+			},
+		}),
+		p4: new Streams.ObjectReadWriteStream({
+			write(data: string) {
+				stream.write(data.replace(/(^|\n)/g, `$1>p4 `));
+			},
+		}),
+	};
 	(async () => {
 		let chunk;
 		// tslint:disable-next-line:no-conditional-assignment
 		while ((chunk = await stream.read())) {
 			const [type, data] = splitFirst(chunk, `\n`);
 			switch (type) {
 			case 'update':
-				const p1Update = data.replace(/\n\|split\n[^\n]*\n([^\n]*)\n[^\n]*\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-				p1.push(p1Update);
-				const p2Update = data.replace(/\n\|split\n[^\n]*\n[^\n]*\n([^\n]*)\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-				p2.push(p2Update);
-				const specUpdate = data.replace(/\n\|split\n([^\n]*)\n[^\n]*\n[^\n]*\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
-				spectator.push(specUpdate);
-				const omniUpdate = data.replace(/\n\|split\n[^\n]*\n[^\n]*\n[^\n]*/g, '');
-				omniscient.push(omniUpdate);
+				const p1Update = data.replace(/\n\|split\n[^\n]*\n([^\n]*)\n[^\n]*\n[^\n]*\n[^\n]*\n[^\n]*/g, '\n$1').replace(/\n\n/g, '\n');
```

Comment:
This part still needs to be reverted, right?

Reply:
I’m sure I already did. Must have been a bad rebase.